mcs_benchmark_data/pipelines/commonsense_qa/commonsense_qa_benchmark_transformer.py

```diff
@@ -23,125 +24,118 @@
 
 class CommonsenseQaBenchmarkTransformer(_Transformer):
 
+    __URI_BASE = "benchmark:commonsense_qa"
+    __BENCHMARK_DATASET_CLASSES = {
+        "dev": BenchmarkDevDataset,
+        "test": BenchmarkTestDataset,
+        "train": BenchmarkTrainDataset,
+    }
+
     def transform(
-        self,
-        *,
-        benchmark_json_file_path: Path,
-        dev_jsonl_file_path: Path,
-        test_jsonl_file_path: Path,
-        train_jsonl_file_path: Path,
-        submission_data_jsonl_file_path: Path,
-        submission_jsonl_file_paths: Tuple[Path, ...]
+        self, benchmark_json_file_path: Path, **kwds
     ) -> Generator[_Model, None, None]:
 
-        benchmark_json = open(benchmark_json_file_path)
-        yield from self.__transform_benchmark(benchmark_json)
-
-        # Yield benchmark dev sample
-        dev_json = open(dev_jsonl_file_path)
-        yield from self.__transform_benchmark_sample(dev_json, "dev")
-
-        # Yield benchmark train sample
-        train_json = open(train_jsonl_file_path)
-        yield from self.__transform_benchmark_sample(train_json, "train")
-
-        # Yield benchmark test sample
-        test_json = open(test_jsonl_file_path)
-        yield from self.__transform_benchmark_sample(test_json, "test")
-
-
-    def __transform_benchmark(self, benchmark_json) -> Generator[_Model, None, None]:
-        # benchmark_bootstrap = BenchmarkBoostrap.from_json(benchmark_json)
```

Comment:
I prefer the approach you had, having an explicit model (BenchmarkBootstrap) for whatever is in the benchmark_json / metadata.json. That model then documents the expected format of that file.

Compare having the model/file format be implicit in the way you read `benchmark_metadata` below. Then you have to document the format of the .json file in some other way.

The former is a common approach: have a model/set of models that exactly mirrors the input format (from the file, database, etc.) and then transform those into other models as necessary (for simpler manipulation in memory, for example).

You could, for example, model the CommonsenseQA JSON lines as a separate @dataclass, CommonsenseQaQuestion. Then you just do CommonsenseQaQuestion.from_json(...) to get the "input format model". Then you transform that model into the ontology models. I find it easier to transform one model (dataclass) to another in Python than to hand-deserialize arbitrary JSON into a model. The IDE can support you better if you're manipulating dataclasses with types, whereas it knows nothing about arbitrary JSON.

Reply:
We'll have to talk about this next week. I am starting to understand the distinction, but part of me still sees this as redundant.

Reply:
It is intentionally redundant -- trading off redundancy for readability. There are many situations where "the simplest thing that could possibly work" decreases readability, and conversely, doing something slightly less simple improves the quality of the code base.

See https://google.github.io/eng-practices/review/reviewer/standard.html

Reply:
Ok sounds good! I appreciate that all of these standards and structures that I learned in Principles of Software are actually used in practice haha.